File: app/src/normal/java/io/agora/education/SettingActivity.java
package io.agora.education;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.CompoundButton;
import android.widget.Switch;
import android.widget.TextView;


import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnCheckedChanged;
import butterknife.OnClick;
import io.agora.edu.launch.AgoraEduSDK;

import static io.agora.education.Constants.KEY_SP;
import static io.agora.education.Constants.POLICYURL;

public class SettingActivity extends AppCompatActivity {
    private static final String TAG = SettingActivity.class.getSimpleName();

    @BindView(R.id.switch_eye_care)
    protected Switch switch_eye_care;
    @BindView(R.id.version_TextView)
    protected TextView versionTextView;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_setting);
        ButterKnife.bind(this);
        switch_eye_care.setChecked(PreferenceManager.get(KEY_SP, false));
        versionTextView.setText(String.format(getString(R.string.version), BuildConfig.VERSION_NAME,
                AgoraEduSDK.version()));
    }

    @OnClick({R.id.iv_back, R.id.layout_policy})
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.iv_back:
                finish();
                break;
            case R.id.layout_policy:
                Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(POLICYURL));
                startActivity(intent);
                break;
            case R.id.layout_log:
//                /**有bug,暂时隐藏不使用*/
//                /**上传log*/
//                getManager().uploadDebugItem(DebugItem.LOG, new EduCallback<String>() {
//                    @Override
//                    public void onSuccess(@Nullable String res) {
//                        Log.e(TAG, "日志上传成功->" + res);
//                    }
//
//                    @Override
//                    public void onFailure(@NotNull EduError error) {
//                        Log.e(TAG, "日志上传错误->code:" + error.getType() + ", reason:" + error.getMsg());
//                    }
//                });
                break;
        }
    }

    @OnCheckedChanged(R.id.switch_eye_care)
    public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
        PreferenceManager.put(KEY_SP, isChecked);
    }

}

File: app/src/main/java/io/agora/education/SettingActivity.kt
package io.agora.education

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.ViewGroup
import android.widget.RelativeLayout
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.AppCompatImageView
import androidx.appcompat.widget.AppCompatTextView

class SettingActivity : AppCompatActivity() {
    private val TAG = "SettingActivity2"
    private lateinit var rootLayout: RelativeLayout
    private lateinit var tvReleaseTime: AppCompatTextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        AppUtil.hideStatusBar(window, true)
        setContentView(R.layout.activity_setting)
        rootLayout = findViewById(R.id.root_Layout)
        val statusBarHeight = AppUtil.getStatusBarHeight(this)
        (rootLayout.layoutParams as ViewGroup.MarginLayoutParams).topMargin = statusBarHeight
        findViewById<AppCompatImageView>(R.id.iv_back).setOnClickListener {
            finish()
        }
        findViewById<RelativeLayout>(R.id.privacyOrdinance).setOnClickListener {
            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(getString(R.string.privacy_url)))
            startActivity(intent)
        }
        findViewById<RelativeLayout>(R.id.termsOfService).setOnClickListener {
            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(getString(R.string.terms_of_service_url)))
            startActivity(intent)
        }
        findViewById<RelativeLayout>(R.id.disclaimers).setOnClickListener {
            val intent = Intent(this, DisclaimersActivity::class.java)
            startActivity(intent)
        }
        findViewById<RelativeLayout>(R.id.registerAgora).setOnClickListener {
            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(getString(R.string.signup_url)))
            startActivity(intent)
        }
        tvReleaseTime = findViewById(R.id.tv_releaseTime)
        tvReleaseTime.text = io.agora.education.BuildConfig.RELEASE_TIME
    }
}